panther_hardware_interfaces/src/panther_system_ros_interface.cpp

```diff
@@ -0,0 +1,138 @@
+                        std::placeholders::_2));
+}
+
+PantherSystemRosInterface::~PantherSystemRosInterface()
+{
+  realtime_driver_state_publisher_.reset();
+  driver_state_publisher_.reset();
+  clear_errors_srv_.reset();
+
+  stop_executor_.store(true);
+  executor_thread_->join();
+  stop_executor_.store(false);
+
+  executor_.reset();
+  node_.reset();
+}
+
+void PantherSystemRosInterface::UpdateMsgErrorFlags(
+  const RoboteqData & front, const RoboteqData & rear)
+{
+  auto & driver_state = realtime_driver_state_publisher_->msg_;
+
+  driver_state.front.fault_flag = front.GetFaultFlag().GetMessage();
+  driver_state.front.script_flag = front.GetScriptFlag().GetMessage();
+  driver_state.front.left_motor_runtime_error = front.GetLeftRuntimeError().GetMessage();
+  driver_state.front.right_motor_runtime_error = front.GetRightRuntimeError().GetMessage();
+
+  driver_state.rear.fault_flag = rear.GetFaultFlag().GetMessage();
+  driver_state.rear.script_flag = rear.GetScriptFlag().GetMessage();
+  driver_state.rear.left_motor_runtime_error = rear.GetLeftRuntimeError().GetMessage();
+  driver_state.rear.right_motor_runtime_error = rear.GetRightRuntimeError().GetMessage();
+}
+
+void PantherSystemRosInterface::UpdateMsgDriversParameters(
+  const DriverState & front, const DriverState & rear)
+{
+  auto & driver_state = realtime_driver_state_publisher_->msg_;
+
+  driver_state.front.voltage = front.GetVoltage();
+  driver_state.front.current = front.GetCurrent();
+  driver_state.front.temperature = front.GetTemperature();
+
+  driver_state.rear.voltage = rear.GetVoltage();
+  driver_state.rear.current = rear.GetCurrent();
+  driver_state.rear.temperature = rear.GetTemperature();
+}
+
+void PantherSystemRosInterface::UpdateMsgErrors(const CANErrors & can_errors)
+{
+  auto & driver_state = realtime_driver_state_publisher_->msg_;
+
+  driver_state.error = can_errors.error;
+  driver_state.write_sdo_error = can_errors.write_sdo_error;
+  driver_state.read_sdo_error = can_errors.read_sdo_error;
+  driver_state.read_pdo_error = can_errors.read_pdo_error;
+
+  driver_state.front.data_timed_out = can_errors.front_data_timed_out;
+  driver_state.rear.data_timed_out = can_errors.rear_data_timed_out;
+
+  driver_state.front.can_net_err = can_errors.front_can_net_err;
+  driver_state.rear.can_net_err = can_errors.rear_can_net_err;
+}
+
+void PantherSystemRosInterface::PublishDriverState()
+{
+  if (realtime_driver_state_publisher_->trylock()) {
+    realtime_driver_state_publisher_->unlockAndPublish();
+  }
+}
+
+void PantherSystemRosInterface::ClearErrorsCb(
+  TriggerSrv::Request::ConstSharedPtr /* request */, TriggerSrv::Response::SharedPtr response)
+{
+  RCLCPP_INFO(node_->get_logger(), "Clearing errors");
+
+  try {
+    clear_errors_();
+    response->success = true;
+  } catch (...) {
+    response->message = "Exception caught in the callback function";
+    response->success = false;
+  }
```

Comment:
```suggestion
  } catch (const std::exception & e) {
    response->message = "Exception caught in the callback function: " + std::string(e.what());
    response->success = false;
  }
```
adds more info why service has failed, p.s. formatting in suggestion might be bad